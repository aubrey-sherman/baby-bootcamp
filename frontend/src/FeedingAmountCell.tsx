import React, { useState, useEffect } from 'react';
import { DateTime } from 'luxon';
import {
  Box,
  Select,
  MenuItem,
  FormControl,
  InputLabel,
  Button,
  Stack,
  Typography,
} from '@mui/material';

type FeedingAmountCellProps = {
  volumeInOunces?: number;
  blockId?: string;
  entryId?: string;
  onAmountSave: (
    blockId: string, entryId: string, newAmount: number
  ) => void;
}

/** Displays and allows editing of the feeding amount.
 *
 * CalendarCell -> FeedingAmountCell
 */
function FeedingAmountCell({
  volumeInOunces = 0,
  blockId,
  entryId,
  onAmountSave
}: FeedingAmountCellProps) {
  const [submitted, setSubmitted] = useState(false);
  const [initialWholeNumber, initialDecimal] = volumeInOunces.toFixed(2).split('.');
  const [wholeNumber, setWholeNumber] = useState(initialWholeNumber);
  const [decimalPart, setDecimalPart] = useState(initialDecimal);

  // Update state when prop changes
  useEffect(() => {
    const [newWholeNumber, newDecimal] = volumeInOunces.toFixed(2).split('.');
    setWholeNumber(newWholeNumber);
    setDecimalPart(newDecimal);
  }, [volumeInOunces]);

  function handleWholeNumberChange(e: any) {
    const value = e.target.value;
    if (value === '' || /^-?\d+$/.test(value)) {
      setWholeNumber(value);
    }
  };

  function handleDecimalChange(e: any) {
    const value = e.target.value;
    if (value === '' || /^\d{0,2}$/.test(value)) {
      setDecimalPart(value.padStart(2, '0'));
    }
  };

  const wholeNumbers = Array.from({ length: 16 }, (_, i) => i);
  const decimalNumbers = Array.from({ length: 4 }, (_, i) =>
    (i * 25).toString().padStart(2, '0')
  );

  const combinedNumber = `${wholeNumber || '0'}.${decimalPart || '00'}`;

  /** Converts saved value from string to number and calls onAmountSave in parent. */
  function handleSubmit(e: React.FormEvent<HTMLFormElement>) {
    e.preventDefault();
    setSubmitted(true);

    const newAmount = Number(combinedNumber);

    // Validate the blockId and entryId instead of using non-null assertion
    if (!blockId || !entryId) {
      console.error('Missing blockId or entryId');
      return;
    }

    // Validate the number before saving
    if (isNaN(newAmount)) {
      console.error('Invalid number format');
      return;
    }

    onAmountSave(blockId!, entryId!, newAmount);
  }

  return (
    <Box sx={{ p: 1 }}>
      <form onSubmit={handleSubmit}>
        <Stack spacing={2}>
          <Typography variant="caption" color="text.secondary">
            Volume in oz
          </Typography>

          <Stack direction="row" spacing={0.5} alignItems="center">
            <FormControl size="small" sx={{ minWidth: 60 }}>
              <InputLabel id="whole-number-label">Oz</InputLabel>
              <Select
                labelId="whole-number-label"
                value={wholeNumber}
                label="Oz"
                onChange={handleWholeNumberChange}
              >
                {wholeNumbers.map(num => (
                  <MenuItem key={num} value={num}>
                    {num}
                  </MenuItem>
                ))}
              </Select>
            </FormControl>

            <Typography variant="h6">.</Typography>

            <FormControl size="small" sx={{ minWidth: 60 }}>
              <InputLabel id="decimal-label">Dec</InputLabel>
              <Select
                labelId="decimal-label"
                value={decimalPart}
                label="Dec"
                onChange={handleDecimalChange}
              >
                {decimalNumbers.map(num => (
                  <MenuItem key={num} value={num}>
                    {num}
                  </MenuItem>
                ))}
              </Select>
            </FormControl>
          </Stack>

          <Button
            type='submit'
            variant="contained"
            size="small"
          >
            Edit amount
          </Button>
        </Stack>
      </form>
    </Box>
  )
}

export default FeedingAmountCell;