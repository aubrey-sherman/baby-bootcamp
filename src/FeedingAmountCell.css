div {
  font-size: 10px;
}

.submit-button {
  background: #4195ff;
  font-size: 12px;
  color: white;
  border: none;
  padding: 1px 2px;
  cursor: pointer;
}