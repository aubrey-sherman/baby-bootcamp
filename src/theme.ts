import { createTheme } from '@mui/material/styles';

const theme = createTheme({
  palette: {
    primary: {
      main: '#0e867a',
    },
    secondary: {
      main: '#fb819b',
    },
    error: {
      main: '#D0021B',
    },
    background: {
      default: '#F5F7FA',
      paper: '#FFFFFF',
    },
    eliminating: {
      main: '#dbeae8',
      contrastText: '#000000',
    },
  },
  typography: {
    fontFamily: '-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, "Helvetica Neue", Arial, sans-serif',
  },
  components: {
    MuiButton: {
      styleOverrides: {
        root: {
          textTransform: 'none',
        },
      },
    },
  },
});

// Extend the palette type to include custom color
declare module '@mui/material/styles' {
  interface Palette {
    eliminating: Palette['primary'];
  }
  interface PaletteOptions {
    eliminating?: PaletteOptions['primary'];
  }
}

export default theme;