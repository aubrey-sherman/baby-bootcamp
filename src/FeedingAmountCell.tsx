import React, { useState, useEffect } from 'react';
import './FeedingAmountCell.css';
import { DateTime } from 'luxon';

type FeedingAmountCellProps = {
  volumeInOunces?: number;
  blockId?: string;
  entryId?: string;
  onAmountSave: (
    blockId: string, entryId: string, newAmount: number
  ) => void;
}

/** Displays and allows editing of the feeding amount.
 *
 * CalendarCell -> FeedingAmountCell
 */
function FeedingAmountCell({
  volumeInOunces = 0,
  blockId,
  entryId,
  onAmountSave
}: FeedingAmountCellProps) {
  const [submitted, setSubmitted] = useState(false);
  const [initialWholeNumber, initialDecimal] = volumeInOunces.toFixed(2).split('.');
  const [wholeNumber, setWholeNumber] = useState(initialWholeNumber);
  const [decimalPart, setDecimalPart] = useState(initialDecimal);

  // Update state when prop changes
  useEffect(() => {
    const [newWholeNumber, newDecimal] = volumeInOunces.toFixed(2).split('.');
    setWholeNumber(newWholeNumber);
    setDecimalPart(newDecimal);
  }, [volumeInOunces]);

  function handleWholeNumberChange(e: React.ChangeEvent<HTMLSelectElement>) {
    const value = e.target.value;
    if (value === '' || /^-?\d+$/.test(value)) {
      setWholeNumber(value);
    }
  };

  function handleDecimalChange(e: React.ChangeEvent<HTMLSelectElement>) {
    const value = e.target.value;
    if (value === '' || /^\d{0,2}$/.test(value)) {
      setDecimalPart(value.padStart(2, '0'));
    }
  };

  const wholeNumbers = Array.from({ length: 16 }, (_, i) => i);
  const decimalNumbers = Array.from({ length: 4 }, (_, i) =>
    (i * 25).toString().padStart(2, '0')
  );

  const combinedNumber = `${wholeNumber || '0'}.${decimalPart || '00'}`;

  /** Converts saved value from string to number and calls onAmountSave in parent. */
  function handleSubmit(e: React.FormEvent<HTMLFormElement>) {
    e.preventDefault();
    setSubmitted(true);

    const newAmount = Number(combinedNumber);

    // Validate the blockId and entryId instead of using non-null assertion
    if (!blockId || !entryId) {
      console.error('Missing blockId or entryId');
      return;
    }

    // Validate the number before saving
    if (isNaN(newAmount)) {
      console.error('Invalid number format');
      return;
    }

    onAmountSave(blockId!, entryId!, newAmount);
  }

  return (
    <div className='FeedingAmountCell'>
      <form onSubmit={handleSubmit}>
        Volume in oz<br />
        <select
          value={wholeNumber}
          onChange={handleWholeNumberChange}
          className="input-right-side"
        >
          {wholeNumbers.map(num => (
            <option key={num} value={num}>
              {num}
            </option>
          ))}
        </select>

        <span className="decimal-point">.</span>

        <select
          value={decimalPart}
          onChange={handleDecimalChange}
          className="input-left-side"
        >
          {decimalNumbers.map(num => (
            <option key={num} value={num}>
              {num}
            </option>
          ))}
        </select>

        <button type='submit' className='submit-button'>
          Edit amount
        </button>
      </form>
    </div>
  )
}

export default FeedingAmountCell;