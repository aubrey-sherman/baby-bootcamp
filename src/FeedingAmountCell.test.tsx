import { describe, it, expect, vi } from 'vitest';
import { render } from '@testing-library/react';
import FeedingAmountCell from './FeedingAmountCell';
import { ThemeProvider } from '@mui/material/styles';
import theme from './theme';

const ThemeWrapper = ({ children }: { children: React.ReactNode }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

describe('FeedingAmountCell', () => {
  const mockOnAmountSave = vi.fn();

  it('renders with existing volume amount', () => {
    const { container } = render(
      <ThemeWrapper>
        <FeedingAmountCell
          volumeInOunces={4.5}
          blockId="block-1"
          entryId="entry-1"
          onAmountSave={mockOnAmountSave}
        />
      </ThemeWrapper>
    );
    expect(container).toMatchSnapshot();
  });

  it('renders with zero volume (new entry)', () => {
    const { container } = render(
      <ThemeWrapper>
        <FeedingAmountCell
          volumeInOunces={0}
          onAmountSave={mockOnAmountSave}
        />
      </ThemeWrapper>
    );
    expect(container).toMatchSnapshot();
  });

  it('renders without blockId or entryId', () => {
    const { container } = render(
      <ThemeWrapper>
        <FeedingAmountCell
          onAmountSave={mockOnAmountSave}
        />
      </ThemeWrapper>
    );
    expect(container).toMatchSnapshot();
  });
});