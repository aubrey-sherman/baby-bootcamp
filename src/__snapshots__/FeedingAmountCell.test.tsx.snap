// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FeedingAmountCell > renders with existing volume amount 1`] = `
<div>
  <div
    class="MuiBox-root css-hpgf8j"
  >
    <form>
      <div
        class="MuiStack-root css-1sazv7p-MuiStack-root"
      >
        <span
          class="MuiTypography-root MuiTypography-caption css-174le52-MuiTypography-root"
        >
          Volume in oz
        </span>
        <div
          class="MuiStack-root css-95rbei-MuiStack-root"
        >
          <div
            class="MuiFormControl-root css-127t2ws-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1htp1mu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              id="whole-number-label"
            >
              Oz
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall MuiSelect-root css-ak97ar-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="whole-number-label"
                class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-xat59s-MuiNativeSelect-root-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                role="combobox"
                tabindex="0"
              >
                4
              </div>
              <input
                aria-hidden="true"
                aria-invalid="false"
                class="MuiSelect-nativeInput css-j0riat-MuiSelect-nativeInput"
                tabindex="-1"
                value="4"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-1rfqz7c-MuiSvgIcon-root-MuiNativeSelect-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-18p5xg2-MuiNotchedOutlined-root-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-ex8a5f-MuiNotchedOutlined-root"
                >
                  <span>
                    Oz
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <h6
            class="MuiTypography-root MuiTypography-h6 css-1z0iuba-MuiTypography-root"
          >
            .
          </h6>
          <div
            class="MuiFormControl-root css-127t2ws-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1htp1mu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              id="decimal-label"
            >
              Dec
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall MuiSelect-root css-ak97ar-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="decimal-label"
                class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-xat59s-MuiNativeSelect-root-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                role="combobox"
                tabindex="0"
              >
                50
              </div>
              <input
                aria-hidden="true"
                aria-invalid="false"
                class="MuiSelect-nativeInput css-j0riat-MuiSelect-nativeInput"
                tabindex="-1"
                value="50"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-1rfqz7c-MuiSvgIcon-root-MuiNativeSelect-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-18p5xg2-MuiNotchedOutlined-root-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-ex8a5f-MuiNotchedOutlined-root"
                >
                  <span>
                    Dec
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary css-14xgzgd-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          Edit amount
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`FeedingAmountCell > renders with zero volume (new entry) 1`] = `
<div>
  <div
    class="MuiBox-root css-hpgf8j"
  >
    <form>
      <div
        class="MuiStack-root css-1sazv7p-MuiStack-root"
      >
        <span
          class="MuiTypography-root MuiTypography-caption css-174le52-MuiTypography-root"
        >
          Volume in oz
        </span>
        <div
          class="MuiStack-root css-95rbei-MuiStack-root"
        >
          <div
            class="MuiFormControl-root css-127t2ws-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1htp1mu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              id="whole-number-label"
            >
              Oz
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall MuiSelect-root css-ak97ar-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="whole-number-label"
                class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-xat59s-MuiNativeSelect-root-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                role="combobox"
                tabindex="0"
              >
                0
              </div>
              <input
                aria-hidden="true"
                aria-invalid="false"
                class="MuiSelect-nativeInput css-j0riat-MuiSelect-nativeInput"
                tabindex="-1"
                value="0"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-1rfqz7c-MuiSvgIcon-root-MuiNativeSelect-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-18p5xg2-MuiNotchedOutlined-root-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-ex8a5f-MuiNotchedOutlined-root"
                >
                  <span>
                    Oz
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <h6
            class="MuiTypography-root MuiTypography-h6 css-1z0iuba-MuiTypography-root"
          >
            .
          </h6>
          <div
            class="MuiFormControl-root css-127t2ws-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1htp1mu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              id="decimal-label"
            >
              Dec
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall MuiSelect-root css-ak97ar-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="decimal-label"
                class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-xat59s-MuiNativeSelect-root-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                role="combobox"
                tabindex="0"
              >
                00
              </div>
              <input
                aria-hidden="true"
                aria-invalid="false"
                class="MuiSelect-nativeInput css-j0riat-MuiSelect-nativeInput"
                tabindex="-1"
                value="00"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-1rfqz7c-MuiSvgIcon-root-MuiNativeSelect-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-18p5xg2-MuiNotchedOutlined-root-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-ex8a5f-MuiNotchedOutlined-root"
                >
                  <span>
                    Dec
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary css-14xgzgd-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          Edit amount
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`FeedingAmountCell > renders without blockId or entryId 1`] = `
<div>
  <div
    class="MuiBox-root css-hpgf8j"
  >
    <form>
      <div
        class="MuiStack-root css-1sazv7p-MuiStack-root"
      >
        <span
          class="MuiTypography-root MuiTypography-caption css-174le52-MuiTypography-root"
        >
          Volume in oz
        </span>
        <div
          class="MuiStack-root css-95rbei-MuiStack-root"
        >
          <div
            class="MuiFormControl-root css-127t2ws-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1htp1mu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              id="whole-number-label"
            >
              Oz
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall MuiSelect-root css-ak97ar-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="whole-number-label"
                class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-xat59s-MuiNativeSelect-root-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                role="combobox"
                tabindex="0"
              >
                0
              </div>
              <input
                aria-hidden="true"
                aria-invalid="false"
                class="MuiSelect-nativeInput css-j0riat-MuiSelect-nativeInput"
                tabindex="-1"
                value="0"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-1rfqz7c-MuiSvgIcon-root-MuiNativeSelect-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-18p5xg2-MuiNotchedOutlined-root-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-ex8a5f-MuiNotchedOutlined-root"
                >
                  <span>
                    Oz
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <h6
            class="MuiTypography-root MuiTypography-h6 css-1z0iuba-MuiTypography-root"
          >
            .
          </h6>
          <div
            class="MuiFormControl-root css-127t2ws-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1htp1mu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              id="decimal-label"
            >
              Dec
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall MuiSelect-root css-ak97ar-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="decimal-label"
                class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-xat59s-MuiNativeSelect-root-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                role="combobox"
                tabindex="0"
              >
                00
              </div>
              <input
                aria-hidden="true"
                aria-invalid="false"
                class="MuiSelect-nativeInput css-j0riat-MuiSelect-nativeInput"
                tabindex="-1"
                value="00"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-1rfqz7c-MuiSvgIcon-root-MuiNativeSelect-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-18p5xg2-MuiNotchedOutlined-root-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-ex8a5f-MuiNotchedOutlined-root"
                >
                  <span>
                    Dec
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary css-14xgzgd-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          Edit amount
        </button>
      </div>
    </form>
  </div>
</div>
`;
